runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1887,11 +1887,42 @@ TR::Register *J9::X86::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::Cod
       auto RDI = cg->allocateRegister();
       auto RCX = cg->allocateRegister();
 
+      TR::Register* tmpReg1 = NULL;
+      TR::Register* tmpReg2 = NULL;
+      TR::Register* tmpXmmYmmReg1 = NULL;
+      TR::Register* tmpXmmYmmReg2 = NULL;
+
+      static bool disableReferenceArrayCopyInlineSmallSizeWithoutREPMOVS  = feGetEnv("TR_DisableReferenceArrayCopyInlineSmallSizeWithoutREPMOVS") != NULL;
+
+      bool enableInlineForSmallSize = !disableReferenceArrayCopyInlineSmallSizeWithoutREPMOVS &&
+                                      !cg->comp()->getOption(TR_DisableReferenceArrayCopyInlineSmallSizeWithoutREPMOVS) &&
+                                      cg->comp()->target().cpu.supportsAVX() &&
+                                      cg->comp()->target().is64Bit();
+
+      int32_t repMovsThresholdBytes = 32;
+      int32_t newThreshold = cg->comp()->getOptions()->getArraycopyRepMovsReferenceArrayThreshold();
+
+      if ((repMovsThresholdBytes < newThreshold) && ((newThreshold == 64) || (newThreshold == 128)))
+         {
+         // If the CPU doesn't support AVX512, reduce the threshold to 64 bytes
+         repMovsThresholdBytes = ((newThreshold == 128) && !cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F)) ? 64 : newThreshold;
+         }
+
+      if (enableInlineForSmallSize)
+         {
+         tmpReg1 = cg->allocateRegister(TR_GPR);
+         tmpReg2 = cg->allocateRegister(TR_GPR);
+         tmpXmmYmmReg1 = cg->allocateRegister(TR_VRF);
+         tmpXmmYmmReg2 = cg->allocateRegister(TR_VRF);
+         }
+
       generateRegRegInstruction(TR::InstOpCode::MOVRegReg(), node, RSI, srcReg, cg);
       generateRegRegInstruction(TR::InstOpCode::MOVRegReg(), node, RDI, dstReg, cg);
       generateRegRegInstruction(TR::InstOpCode::MOVRegReg(), node, RCX, sizeReg, cg);
 
-      auto deps = generateRegisterDependencyConditions((uint8_t)5, 5, cg);
+      TR::RegisterDependencyConditions* deps = enableInlineForSmallSize ? generateRegisterDependencyConditions((uint8_t)9, 9, cg) :
+                                                                          generateRegisterDependencyConditions((uint8_t)5, 5, cg);
```

Comment:
You can keep the generated code smaller by:
```
   int8_t numDeps = enableInlineForSmallSize ? 9 : 5;
   TR::RegisterDependencyConditions *deps = generateRegisterDependencyConditions(numDeps, numDeps, cg);
```

Reply:
Addressed in [e721562](https://github.com/eclipse-openj9/openj9/compare/21255244442efda857a4a2c7772e1c6204a8d084..e7215625c5684fa2a4397eb5d65d1bf3a3eb8c60)